chain/epoch_manager/src/validator_selection.rs

```diff
@@ -51,11 +51,25 @@ pub fn proposals_to_epoch_info(
     let (block_producers, bp_stake_threshold) =
         select_block_producers(&mut block_producer_proposals, max_bp_selected, min_stake_ratio);
     let mut chunk_producer_proposals = order_proposals(proposals.into_iter().map(|(_, p)| p));
-    let (chunk_producers, cp_stake_threshold) = select_chunk_producers(
-        &mut chunk_producer_proposals,
-        max_cp_selected,
-        min_stake_ratio,
-        num_shards,
+    let (chunk_producers, cp_stake_threshold) = checked_feature!(
+        "protocol_feature_chunk_only_producers",
+        ChunkOnlyProducers,
+        next_version,
+        {
+            let max_cp_selected = max_bp_selected
+                + (epoch_config.validator_selection_config.num_chunk_only_producer_seats as usize);
+            select_chunk_producers(
+                &mut chunk_producer_proposals,
+                max_cp_selected,
+                min_stake_ratio,
+                num_shards,
+            )
+        },
+        {
+            let _ = chunk_producer_proposals;
```

Comment:
if we don't use chunk_producer_proposals -> maybe we should move the line 53 into 59? (so that order_proposals is computed only when we do the chunk_only producers?